libs/redis/files/redis.init

```diff
@@ -0,0 +1,21 @@
+#!/bin/sh /etc/rc.common
+
+USE_PROCD=1
+START=95
+STOP=10
+
+REDIS_BIN="/usr/bin/redis-server "
+REDIS_CLI="/usr/bin/redis-cli"
+REDIS_CONFIG="/etc/redis.conf"
+REDIS_PID="/var/run/redis.pid"
+REDIS_DATA="/opt/var/lib/redis"
+REDIS_SERVER="/usr/bin/redis-server"
+
+start_service() {
+	mkdir -p "$REDIS_DATA"
+	rm -rf /var/lib/redis || true
+	ln -s "$REDIS_DATA" /var/lib/redis || true
+	procd_open_instance redis
+	procd_set_param command "$REDIS_SERVER" "$REDIS_CONFIG" --dir "$REDIS_DATA"
```

Comment:
I would suggest simply leave out the `--dir` part to let the config file have a final say.  It should be actually more flexible and safer because we do not have to do `rm -rf`

Reply:
"ROM" -> "RAM".

Users will need to be aware of the storage kind and size requirement no matter what.  Saving it by default on some non-volatile persistent local storage is no better than the default /var/lib/redis, as the size constraint is still there with the additional effect that periodic dump will wear out possibly nor flash.

Whether to save the dump, how and where to save it are configurable, that's why I think we should not hard code the logic in the code.  Too many possibilities to do it right.

Reply:
How about `$(INSTALL_DIR)` $(1)/opt/redis` and set `dir /opt/redis` in the default redis.conf

Reply:
Well, I hope we both made our points clear ;)

Mine is about scripting the behavior in the initscript or letting the config file have the final say.

Reply:
I was thinking about that and decided to set the default data directory to /var/lib/redis . The reason for that is that if someone runs redis just out of curiosity, it will not damage permanent storage by writing to it. In case someone wants to change that behavior it is still possible to edit dir inside /etc/redis.conf to permanent storage.